tasks/types.hh

```diff
@@ -14,4 +14,16 @@ namespace tasks {
 
 using task_id = utils::tagged_uuid<struct task_id_tag>;
 
+struct task_info {
+    task_id id;
+    unsigned shard;
+
+    task_info() noexcept : id(task_id::create_null_id()) {}
+    task_info(task_id parent_id, unsigned parent_shard) noexcept : id(parent_id), shard(parent_shard) {}
```

Comment:
now that the class is not necessarily about parent, maybe the constructor params should not mention parent, as I guess this would be part of the context of the constructed object.